### PR TITLE
[GSK-1061] Catalog tab: UI improvements

### PR DESCRIPTION
**Pull request by henchaves**
## Description
This PR aims to polish the Catalog tab and its subtabs.
A lot of updates were implemented:
- Move Datasets and Models to Catalog
- Improve Description/Inputs/Code sections of Tests/Slicing Funcs/Transformation Funcs
- Add code examples of how to use each Test

## Related Issue
[GSK-1061 (available on Linear)](https://linear.app/giskard/issue/GSK-1061/catalog-tab-ui-improvements)

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [x] 🥂 Improvement (non-breaking change which improves an existing feature)
- [x] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

### CODE REVIEW

File: frontend/src/views/main/project/Datasets.vue [hunk 1]
```diff
@@ -61,6 +63,28 @@
       <CodeSnippet :code-content="codeContent" :language="'python'"></CodeSnippet>
       <p class="mt-4 font-weight-medium secondary--text">Check out the <a href="https://docs.giskard.ai/start/~/changes/QkDrbY9gX75RDMmAWKjX/guides/upload-your-model#3.-create-a-giskard-dataset" target="_blank">full documentation</a> for more information.</p>
     </v-container>
+
+    <!-- Upload dialog -->
+    <v-dialog v-model="openDialog" max-width="800px">
```
Comment (kevinmessiaen) [inline frontend/src/views/main/project/Datasets.vue:68]: We should use the `$vfm.show` method to show modal programmatically

File: frontend/src/views/main/project/Models.vue [hunk 2]
```diff
@@ -72,6 +74,27 @@
       <CodeSnippet :code-content="codeContent" :language="'python'"></CodeSnippet>
       <p class="mt-4 font-weight-medium secondary--text">Check out the <a href="https://docs.giskard.ai/start/~/changes/QkDrbY9gX75RDMmAWKjX/guides/upload-your-model#2.-create-a-giskard-model" target="_blank">full documentation</a> for more information.</p>
     </v-container>
+
+    <!-- Upload dialog -->
+    <v-dialog v-model="openDialog" max-width="800px">
```
Comment (kevinmessiaen) [inline frontend/src/views/main/project/Models.vue:79]: Same here we should have `$vfm.show`

File: frontend/src/views/main/utils/DatasetColumnSelector.vue [hunk 3]
```diff
@@ -53,6 +43,6 @@ const availableColumns = computed(() => {
 <style scoped>
 .slice-function-selector {
     min-width: 200px;
-    flex-grow: 1;
+    /* flex-grow: 1; */
```
Comment (kevinmessiaen) [inline frontend/src/views/main/utils/DatasetColumnSelector.vue:46]: We can delete this row

```suggestion
```

File: frontend/src/views/main/utils/SlicingFunctionSelector.vue [hunk 4]
```diff
@@ -1,20 +1,11 @@
...
-            item-value="uuid"
-            :return-object="false"
-            @input="onInput"
-            :dense="fullWidth"
-            hide-details
-            :prepend-inner-icon="icon ? 'mdi-filter' : null"
-        ></v-select>
+    <div class="d-flex" :class="{ w100: fullWidth }">
+        <v-select clearable :outlined="fullWidth" class="slice-function-selector" :label="label" v-model="value" :items="[{
+            name: 'None',
```
Comment (kevinmessiaen) [inline frontend/src/views/main/utils/SlicingFunctionSelector.vue:4]: I don't understand why we have a `None` argument being passed

**Review by kevinmessiaen — CHANGES_REQUESTED** [review]

File: frontend/src/views/main/utils/SlicingFunctionSelector.vue (hunk 4 above)
Comment (henchaves) [reply frontend/src/views/main/utils/SlicingFunctionSelector.vue:4]: I and @jmsquare took the decision to display the default value for each input (if it has a default value). 
In the case of the Slicing Function, it was displaying a selector with no option selected, so it was not very clear if it was a required field or not.

**Pull request merged**